python/tvm/relay/frontend/onnx.py

```diff
@@ -873,6 +873,89 @@ def flatten_to_nd(x, x_shape, nd=3):
         return _op.nn.dense(inputs[0], input_1_t)
 
 
+class MatMulInteger16(OnnxOpConverter):
+    """Operator converter for MatMulInteger16 from Microsoft onnxruntime contrib opset."""
+
+    @classmethod
+    def _impl_v10(cls, inputs, attr, params):
+        assert len(inputs) == 2, "MatMul op take 2 inputs, {} given".format(len(inputs))
+        a_shape = shape_of(inputs[0])
+        a_rank = infer_shape(a_shape)[0]
+        b_shape = shape_of(inputs[1])
+        b_rank = infer_shape(b_shape)[0]
+        a_dtype = infer_type(inputs[0]).checked_type.dtype
+        b_dtype = infer_type(inputs[1]).checked_type.dtype
+        # Check input data types
+        assert a_dtype in ("int16", "uint16"), "MatMulInteger16: invalid dtype for first input"
+        assert b_dtype in ("int16", "uint16"), "MatMulInteger16: invalid dtype for second input"
+        out_dtype = "int32"
+        # Set output data type as uint32 when both inputs are uint16
+        if a_dtype == "uint16" and b_dtype == "uint16":
+            out_dtype = "uint32"
+        if a_rank > 2 or b_rank > 2:
+
+            def flatten_to_nd(x, x_shape, nd=3):
```

Comment:
nitpick: All 3 calls to this function explicitly specify the `nd` value. Does it still make sense to provide a default value here?

Reply:
I reused this function from `batch_matmul`. 
An alternative to this could be to not set `nd `value while calling `flatten_to_nd`

Reply:
Would it possibly make sense to move that function outside of `batch_matmul`, and then you could just call that existing implementation?  It would be nice to avoid duplicating that much code.

Reply:
Yes. Agree with you. I will fix it in the next commit.

Reply:
This is fixed in the following commit: `ccbe433937c0661414a5e62e1f6321058057e158`